django/db/models/sql/query.py

```diff
@@ -1708,6 +1708,8 @@ def _gen_cols(cls, exprs):
         for expr in exprs:
             if isinstance(expr, Col):
                 yield expr
+            elif isinstance(expr, F):
+                yield expr.name
```

Comment:
This should return a `Col()` instance, I will fix it later.

Reply:
Hmm that's strange, this branch should not be necessary assuming `exprs` all are resolved before hand.

Reply:
@charettes they are resolved locally in `create_sql`. We could return the resolved `ExpressionList` in a cached_property like `Index.resolved_expressions`.

Reply:
I'm up for anything that avoids leaking that `F` resolves to `Col` in `sql.Query`.

Reply:
This is not straightforward, because resolving expressions with nonexistent fields throws `FieldError`. I think we can handle this separately, I added ticket-32321.